### PR TITLE
Add _mut methods for edge mutation for faster single-thread access

### PR DESCRIPTION
**Pull request by s1ck**

### CODE REVIEW

File: crates/builder/src/graph/adj_list.rs [hunk 1]
```diff
@@ -71,6 +71,33 @@ impl<NI: Idx, EV> AdjacencyList<NI, EV> {
...
+            CsrLayout::Sorted => match edges.binary_search(&target) {
+                Ok(i) => edges.insert(i, target),
+                Err(i) => edges.insert(i, target),
+            },
+            CsrLayout::Unsorted => edges.push(target),
+            CsrLayout::Deduplicated => match edges.binary_search(&target) {
+                Ok(_) => {}
+                Err(i) => edges.insert(i, target),
+            },
+        };
```
Comment (s1ck) [inline crates/builder/src/graph/adj_list.rs:89]: @knutwalker this is duplicated in the method above. I was wondering if we can write this generic to work on `&mut Vec<Target<NI, EV>>` and `RwLockWriteGuard<Vec<Target<NI, EV>>>`

Comment (knutwalker) [reply crates/builder/src/graph/adj_list.rs:89]: @s1ck `RwLockWriteGuard<Vec<Target<NI, EV>>>` should deref to a `&mut Vec<Target<NI, EV>>` so can implement one in terms of the other.

Comment (knutwalker) [reply crates/builder/src/graph/adj_list.rs:89]: @s1ck `RwLockWriteGuard<Vec<Target<NI, EV>>>` should deref to a `&mut Vec<Target<NI, EV>>` so can implement one in terms of the other.

Comment (knutwalker) [reply crates/builder/src/graph/adj_list.rs:89]: @s1ck `RwLockWriteGuard<Vec<Target<NI, EV>>>` should deref_mut to a `&mut Vec<Target<NI, EV>>` so can implement one in terms of the other.

Comment (s1ck) [reply crates/builder/src/graph/adj_list.rs]: right, I tried that but got mislead by another borrow problem. works now! thx

**Pull request merged**